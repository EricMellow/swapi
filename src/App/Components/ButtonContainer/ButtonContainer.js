import React from 'react';
import Button from '../Button/Button.js';
import './ButtonContainer.css'

const ButtonContainer = ({getPeople, getPlanets, getStarships, getVehicles}) => {
  return (
    <div className='btn-container'>
      <Button 
        name={'People'}
        handleButtonClick={getPeople}
      />
      <Button
        name={'Planets'}
        handleButtonClick={getPlanets}
      />
      <Button
        name={'Starships'}
        handleButtonClick={getStarships}
      />
      <Button
        name={'Vehicles'}
        handleButtonClick={getVehicles}
      />
    </div>
);
};

export default ButtonContainer;