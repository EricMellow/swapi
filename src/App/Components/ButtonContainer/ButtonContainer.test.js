import React from "react";
import ButtonContainer from './ButtonContainer';
import { shallow } from "enzyme";

describe('ButtonContainer', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<ButtonContainer />);
    
    expect(wrapper).toMatchSnapshot();
  });
});