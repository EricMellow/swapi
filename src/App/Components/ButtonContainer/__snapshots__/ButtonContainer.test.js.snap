// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/people"
>
          People
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/planets"
>
          Planets
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/vehicles"
>
          Vehicles
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="button"
          onClick={[Function]}
          to="/starships"
>
          Starships
</NavLink>,
      ],
      "className": "btn-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "People",
          "className": "button",
          "onClick": [Function],
          "to": "/people",
        },
        "ref": null,
        "rendered": "People",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Planets",
          "className": "button",
          "onClick": [Function],
          "to": "/planets",
        },
        "ref": null,
        "rendered": "Planets",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Vehicles",
          "className": "button",
          "onClick": [Function],
          "to": "/vehicles",
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Starships",
          "className": "button",
          "onClick": [Function],
          "to": "/starships",
        },
        "ref": null,
        "rendered": "Starships",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/people"
>
            People
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/planets"
>
            Planets
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/vehicles"
>
            Vehicles
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="button"
            onClick={[Function]}
            to="/starships"
>
            Starships
</NavLink>,
        ],
        "className": "btn-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "People",
            "className": "button",
            "onClick": [Function],
            "to": "/people",
          },
          "ref": null,
          "rendered": "People",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Planets",
            "className": "button",
            "onClick": [Function],
            "to": "/planets",
          },
          "ref": null,
          "rendered": "Planets",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Vehicles",
            "className": "button",
            "onClick": [Function],
            "to": "/vehicles",
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Starships",
            "className": "button",
            "onClick": [Function],
            "to": "/starships",
          },
          "ref": null,
          "rendered": "Starships",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
